src/server/apsystem.ts

```diff
@@ -374,12 +374,39 @@ export default class ActivityPubSystem {
       await this.hookSystem.dispatchOnApproved(fromActor, activity)
     } else if (type === 'Undo') {
       await this.performUndo(fromActor, activity)
-    } else {
+    } else if ((type === 'Create') || (type === 'Update')) {
+      if (typeof activity.object === 'string') {
+        const response = await this.signedFetch(fromActor, {
+          method: 'get',
+          url: activity.object,
+          headers: {
+            'Content-Type': 'application/ld+json'
+          }
+        })
+
+        if (!response.ok) { throw createError(404, `Unable to load object for activity at ${activity.object}`) }
+
+        const object = await response.json()
+        // We check that the activity actor is set elsewhere
+        await this.storeObject(fromActor, object, activity.actor as string)
```

Comment:
i think you need to check if the fetched object belongs to the actor, something like `object.attributedTo === activity.actor`

Reply:
or have a helper function to perform these checks, i imagine we'll need them everywhere

Reply:
also check the id referenced and retrieved match

Reply:
I do this check inside storeObject, s that sufficient?

Reply:
> also check the id referenced and retrieved match

this too?